amanuensis/resources/userdatamodel/userdatamodel_project.py

```diff
@@ -58,34 +96,67 @@ def create_project(current_session, user_id, description, name, institution, sea
     # current_session.flush()
     # current_session.refresh(new_project)
     current_session.commit()
-    
+
     return new_project
 
 
 def update_project(current_session, project_id, approved_url=None, searches=None):
     if not approved_url and not searches:
-        return {"code": 200, "error": "Nothing has been updated, no new values have been received by the function."}
+        return {
+            "code": 200,
+            "error": "Nothing has been updated, no new values have been received by the function.",
+        }
 
     data = {}
     if approved_url:
-        data['approved_url'] = approved_url
+        data["approved_url"] = approved_url
     if searches and isinstance(searches, list) and len(searches) > 0:
         data["searches"] = searches
 
-    #TODO check that at least one has changed
-    num_updated = current_session.query(Project).filter(
-        Project.id == project_id
-    ).update(data)
-    if  num_updated > 0:
-        return  {"code": 200, "updated": int(project_id)}
+    # TODO check that at least one has changed
+    num_updated = (
+        current_session.query(Project).filter(Project.id == project_id).update(data)
+    )
+    if num_updated > 0:
+        return {"code": 200, "updated": int(project_id)}
     else:
-        return {"code": 500, "error": "Nothing has been updated, check the logs to see what happened during the transaction."}
+        return {
+            "code": 500,
+            "error": "Nothing has been updated, check the logs to see what happened during the transaction.",
+        }
 
 
 def get_statisticians(current_session, emails):
     if not emails:
         return []
-    return current_session.query(Statistician).filter(Statistician.email.in_(emails)).all()
+    return (
+        current_session.query(Statistician).filter(Statistician.email.in_(emails)).all()
+    )
+
+
+def update_project_date(session, project_id, new_update_date):
+    requests = get_requests_by_project_id(session, project_id)
+    if not requests:
+        raise NotFound(
+            "There are no requests associated to this project or there is no project. id: {}".format(
+                project_id
+            )
+        )
+
+    for request in requests:
+        create_date = request.request_has_state[-1].create_date
+        if create_date <= new_update_date:
+            request.update_date = new_update_date
```

Comment:
Why do you compare with the create date instead of the update_date?

Reply:
Because I thought we didn't want the new date date to come before the date the request was created.